libdevcore/RLP.cpp

```diff
@@ -149,7 +149,7 @@ unsigned RLP::length() const
 		return n - c_rlpDataImmLenStart;
 	else if (n < c_rlpListStart)
 	{
-		if ((int)m_data.size() <= n - c_rlpDataIndLenZero)
+		if ((int)m_data.size() <= n - c_rlpDataIndLenZero || m_data[1] == 0)
```

Comment:
When I send you the wrong "RLP" : 0xb8, you will arrive here, and just end with a segfault, because m_data[1] does not exist. This means, when this PR is merged, I can send a malicous transaction with 0xb8 and crash the client.